### ISSUE TITLE
[v3.7.0] (Regression) Rigidbody2D using Box2D does not update its position

### ISSUE THREAD

**Issue by mrtzd** [cocos/cocos-engine#14209]
### Cocos Creator version

3.7.0

### System information

Windows 10 Editor

### Issue description

This was not an issue with v3.5.2 or v3.6.3. This only happens with v3.7.0.

I have attached a node to a socket of a dragonbones skeleton (like below) and I have a RigidBody2D on the attached node. The problem is that when I move the the bone, the RigidBody2D of the attached node does not move with the socket node.

![image](https://user-images.githubusercontent.com/82378750/218344760-aabe9c71-332a-4112-a865-44d21c485337.png)

Here's a sample code to test this issue:
```
private armatureDisplay: dragonBones.ArmatureDisplay;
private armature: dragonBones.Armature;
private bone: dragonBones.Bone;
private time: number = 0;
private direction: number = 1;

start() {
    PhysicsSystem2D.instance.debugDrawFlags = EPhysics2DDrawFlags.Shape;
    this.armatureDisplay = this.node.getComponent(dragonBones.ArmatureDisplay);
    this.armature = this.armatureDisplay.armature();
    this.bone = this.armature.getBone('RightWing');
}

update(deltaTime: number) {
    this.time += deltaTime;
    if (this.time > 1) {
        this.time = 0;
        this.direction *= -1;
        this.bone.offset.rotation = misc.degreesToRadians(20 * this.direction);
        this.bone.invalidUpdate();
    }
}
```

Here's how it looks in game (both debug bones and physics flags are enabled):

https://user-images.githubusercontent.com/82378750/218344746-b1ab6013-2fa2-477e-9199-968679a2cfb9.mp4

### Steps to reproduce

Attach a node with RigidBody2D to a socket on a DragonBones skeleton and then enable debug flags for physics to see if the rigidbody is moving with the socket node.

Update: You have to use the Box2D physics system.

### Minimal reproduction project

[Demo2D.zip](https://github.com/cocos/cocos-engine/files/10717665/Demo2D.zip)

**Comment by mrtzd** [comment]
I've looked into this issue a little bit more and I found out that this problem only happens when using Box2D physics system. When I switched to the Builtin physics system the RigidBody2D correctly updates its position with the socket node like this:

https://user-images.githubusercontent.com/82378750/218749015-6b37b09a-6079-48d7-9b33-eb7318572c3e.mp4

I still need to solve this issue since I need to use Box2D physics for my game and this was working correctly with previous versions (v3.6.3 and v3.5.2).

**Comment by mrtzd** [comment]
Ok I found where this regression was introduced, I'll make a pr for this.

**Comment by mrtzd** [comment]
Fixed by #14162
